samples/18.customization-open-url/README.md

```diff
@@ -0,0 +1,111 @@
+# Sample - Customize open URL behavior
+
+Web Chat client that will show a confirmation dialog when opening an URL.
+
+# Test out the hosted sample
+
+- [Try out MockBot](https://microsoft.github.io/BotFramework-WebChat/18.customization-open-url)
+
+# Things to try out
+
+- Type "card sports" in the send box
+- Click on the "Seattle vs Panthers" card
+
+# Code
+
+> Jump to [completed code](#completed-code) to see the end-result `index.html`.
+
+### Goals of this bot
+
+The `index.html` page has one main goal:
+
+- To intercept all open URL actions
+
+We'll start by using the [full-bundle CDN sample](./../01.a.getting-started-full-bundle/README.md) as our Web Chat template.
+
+```diff
+…
+window.WebChat.renderWebChat({
+  directLine: window.WebChat.createDirectLine({ token }),
++ openUrlPonyfillFactory: ({ cardAction: { type } }) => {
++   switch (type) {
++     case 'signin':
++       const popup = window.open();
++
++       return url => popup.location.href = url;
++
++     default:
++       return url => {
++         if (confirm(`Do you want to open this URL?\n\n${ url }`)) {
++           window.open(url, '_blank');
++         }
++       };
++   }
+  },
```

Comment:
This doesn't really feel like a ponyfill, its more of a custom url click handler